### ISSUE TITLE
remember search term

### ISSUE THREAD

**Issue by TomTomH2** [peterdotjs/quick-find#15]
Could you please add an option to remember the search term when using ctrl-F on multiple (consecutive) pages.

It is most annoying having to re-type your search term each and every time one uses ctrl-F on consecutive web-pages, stemming for instance from consecutive "hits" from a google-search.

Amazingly enough I've found a bunch of tools on the web enabling the search of multiple search-terms with AND/OR on a single web-page but almost nothing for finding one term on multiple pages without re-typing it in every new instance.

many thanks in advance
TomTomH2

**Comment by peterdotjs** [comment]
@TomTomH2 I don't think it would make sense to autofill the search box across tabs/pages but there was a feature I thought of before #12 where previously searched results would popup in the menu first so it would ease the need to copy paste/retype recent searches.

**Comment by TomTomH2** [comment]
It's your program - but to me it would make extreme sense, all the more if
this behavior could be toggled as an option.

Take for instance a food-home-delivery-service-page. You use the
page-search-function to find restaurants that deliver eggs of different
kind (or toast or whatever) and now you get 20 restaurants as a result.

What follows next is, you have to open each and every one of them to see in
which form they offer their eggs (search term). This is when a local
find-dialogue is needed to jump from one occurrence of "eggs" to the next.

If you now open each of the 20 restaurants menu-pages on a new page/Tab to
find all occurrences of "eggs", it would be extremely cumbersome having to
re-type "eggs" 20 times for each new page/Tab into the local search-mask.

I would rather prefer to have "eggs" pre-entered in the search-mask on all
20 pages (selected in blue) until I replace it with a new search term by
simply typing the new word.

Hope, that helps to illustrate my problem.

Thanks and greetings
TomTom

**Comment by peterdotjs** [comment]
@TomTomH2 I do see your use case but I hope you understand that it is a very specific feature request. I'll keep this open as an enhancement request for now. Usage is pretty low for this extensions only ~1k weekly active users so I really appreciate you providing the feedback in helping me understand how people are using it. I've been pretty busy with other stuff so I'm not sure when I'll get to it but feel free to make a pull request if you're interested in the meantime.

Thanks!
Peter

**Comment by peterdotjs** [comment]
Details TBD: How to determine previous search term. There may be some difficulty in making this determination - some criteria/requirements needs to be specified.

**Comment by TomTomH2** [comment]
How to determine "previous search term", you ask? Well, it couldn't be
easier. The last word/term, that has been typed into
the box/search_mask before, would stay permanently in the box/search_mask -
until it is replaced by a new one.

I also fully support your idea of a drop-down-list of "recent search terms"
to choose from. Technically speaking you would have to buffer those recent
search terms into a file associated with the extension. And the
last/top_entry in that list would then automatically become the
"permanent"/pre_selected/active entry in the search-mask - until it is
replaced by another one either from the drop_down-list or by typing in a
new term.

An option (maybe called "sticky search term") could then decide, whether or
not a new search would start with the last/most_recent search term from the
list as active/pre_selected entry - or with an empty box instead.

That - in my mind - would be the perfect solution.

greetings
TomTom

**Comment by TomTomH2** [comment]
I am afraid I do NOT see my request as a "very specific use case" - and
here is why.

Forget about food. I'll give you another example. You do a google-search
for a technical issue, let's say "direct methanol fuel cell".

As a result google comes up with a list of 250000 hits, the top_20 you now
would like to browse through. Of course you do not have the time to read
them altogether and in detail.

Again the local "Find-mask" comes into play. As you open each of those
top_20-google-hits in a new window you would now want to jump from one
occurrence of "direct methanol fuel cell" WITHIN each article to the next -
just to quickly determine, which of those 20+ articles seems to be the most
relevant with respect to your initial query.

So this is about your typical, everyday-style google search of arbitrary
search terms. And believe me it is much more cumbersome to 20-times re-type
"direct methanol fuel cell" as compared to "eggs". ;-)

And perhaps the relatively low use of "ctrl-F" in general is mainly founded
in the very limited ability of the built-in standard tool to really assist
the user with his typical everyday-search-problem in an adequate manner.

greetings
TomTom

**Comment by peterdotjs** [comment]
@TomTomH2 from a technical standpoint requirements need to be made to determine what is the "search term to be saved". Every key entered is considered a search term with an existing timeout. The comment was more of a placeholder TODO. It would not be performant to save the search terms in a trivial manner. 

As per your use case why not use copy paste? "eggs" is the same effort as "direct methanol fuel cell". I reiterate that it is a specific use case as of now since no one else has given a +1. If more people give support for it then it'll prove it isn't a special case.

And as a I said before if you do believe it to be a very important feature you are free to make a pull request for these changes.

**Comment by TomTomH2** [comment]
I am afraid, I really do not get what you mean by requirements in
connection to the "search term to be saved". In "my" new sticky mode every
final search term/string (when typing is done) that has been searched once
throughout one page would automatically become top_of_the_list and "sticky"
for the next searches - until it is replaced by a new key-entry.

Even the standard "ctrl-F" feature allows for keeping the last search term
throughout the next consecutive pages by pressing the "F3"-button.

And I really can't believe that you do not have the same problem yourself
in your own google searches. How on earth would you sift through your own
google search-results in a much different way than I do (or millions of
other users, for that purpose)? And of course you/I/everyone uses
"copy&paste" to avoid re-typing (if you do not forget to copy at the first
time). But doesn't it bother you at all yourself having to paste the same
term 20+ times and re-evoking the search on each new page you want to sift
through? For me that consists a major nuisance and the "F3"-mode in
conventional "ctrl-F" is a godsend.

As for the "specific use case"-matter I will admit that I am not much into
the "+1-game" and only have a vague idea what a "pull request" is and how
to make one within the forum. I'm not much of a "forum-guy" and had rather
hoped you would see the benefit of my idea for yourself immediately.

As I sad, in the end it is up to you entirely what you make of my idea.

greetings
TomTom
